build/jobs/provision-deploy.yml

```diff
@@ -128,6 +128,28 @@ jobs:
             }
           }
         }
+
+        if("${{ parameters.sql }}" -eq "false"){
+            Write-Host "Add DevOps MI permission to Cosmos databases"
+
+            $principalId = (Get-AzContext).Account.Id
+
+            New-AzCosmosDBSqlRoleAssignment `
+              -AccountName $webAppName `
+              -ResourceGroupName $resourceGroupName `
+              -Scope "/" `
+              -PrincipalId $principalId `
```

Comment:
Currently failing here since the Federated Principal doesn't exist in the same tenant as the cosmos db

Reply:
Solved, thanks @mikaelweave :)